src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionProvider.cs

```diff
@@ -78,79 +82,122 @@ private async Task AddCompletionItemsAsync(CompletionContext completionContext,
-                    if (assemblyProject != null && assemblyProject.SupportsCompilation)
-                    {
-                        await typeImportCompletionService.GetTopLevelTypesAsync(
-                            assemblyProject,
-                            GetHandler(compilation.Assembly, assembly),
-                            cancellationToken).ConfigureAwait(false);
-                    }
-                    else if (metadataReference is PortableExecutableReference peReference)
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    // Skip reference with only non-global alias.
+                    var metadataReference = compilation.GetMetadataReference(referencedAssemblySymbol);
+
+                    if (metadataReference.Properties.Aliases.IsEmpty ||
+                        metadataReference.Properties.Aliases.Any(alias => alias == MetadataReferenceProperties.GlobalAlias))
                     {
-                        typeImportCompletionService.GetTopLevelTypesFromPEReference(
-                            project.Solution,
-                            compilation,
-                            peReference,
-                            GetHandler(compilation.Assembly, assembly),
-                            cancellationToken);
+                        var assemblyProject = project.Solution.GetProject(referencedAssemblySymbol, cancellationToken);
+                        if (assemblyProject != null && assemblyProject.SupportsCompilation)
+                        {
+                            return await typeImportCompletionService.GetTopLevelTypesAsync(assemblyProject,
+                                                                                        syntaxContext,
+                                                                                        GetHandler(compilation.Assembly, referencedAssemblySymbol),
+                                                                                        delayItemHandling: true,
+                                                                                        cancellationToken).ConfigureAwait(false);
+                        }
+
+                        if (metadataReference is PortableExecutableReference peReference)
+                        {
+                            return Task.Run<Task>(() => typeImportCompletionService.GetTopLevelTypesFromPEReference(
+                                project.Solution,
+                                compilation,
+                                peReference,
+                                syntaxContext,
+                                GetHandler(compilation.Assembly, referencedAssemblySymbol),
+                                delayItemHandling: true,
+                                cancellationToken), cancellationToken);
+                        }
                     }
+
+                    return Task.CompletedTask;
```

Comment:
this feels super weird.  returning a completed-task from an already async method...

Reply:
Fixed